test/e2e/tests/single_zone_e2e_test.go

```diff
@@ -539,4 +496,110 @@ var _ = Describe("GCE PD CSI Driver", func() {
 			Expect(gce.IsGCEError(err, "notFound")).To(BeTrue(), "Expected snapshot to not be found")
 		}()
 	})
+
+	It("Should get correct VolumeStats for Block", func() {
+		testContext := getRandomTestContext()
+
+		p, z, _ := testContext.Instance.GetIdentity()
+		client := testContext.Client
+		instance := testContext.Instance
+
+		volName, volID := createAndValidateUniqueZonalDisk(client, p, z)
+
+		defer func() {
+			// Delete Disk
+			err := client.DeleteVolume(volID)
+			Expect(err).To(BeNil(), "DeleteVolume failed")
+
+			// Validate Disk Deleted
+			_, err = computeService.Disks.Get(p, z, volName).Do()
+			Expect(gce.IsGCEError(err, "notFound")).To(BeTrue(), "Expected disk to not be found")
+		}()
+
+		verifyVolumeStats := func(a verifyArgs) error {
+			available, capacity, used, inodesFree, inodes, inodesUsed, err := client.NodeGetVolumeStats(volID, a.publishDir)
+			if err != nil {
+				return fmt.Errorf("failed to get node volume stats: %v", err)
+			}
+			if available != 0 || capacity != common.GbToBytes(defaultSizeGb) || used != 0 ||
+				inodesFree != 0 || inodes != 0 || inodesUsed != 0 {
+				return fmt.Errorf("got: available %v, capacity %v, used %v, inodesFree %v, inodes %v, inodesUsed %v -- expected: capacity = %v, available = 0, used = 0, inodesFree = 0, inodes = 0 , inodesUsed = 0",
+					available, capacity, used, inodesFree, inodes, inodesUsed, common.GbToBytes(defaultSizeGb))
+			}
+			return nil
+		}
+
+		// Attach Disk
+		err := testLifecycleWithVerify(volID, volName, instance, client, false /* readOnly */, true /* block */, verifyVolumeStats, nil)
+		Expect(err).To(BeNil(), "Failed to go through volume lifecycle")
+	})
+
+	It("Should get correct VolumeStats", func() {
+		testContext := getRandomTestContext()
+
+		p, z, _ := testContext.Instance.GetIdentity()
+		client := testContext.Client
+		instance := testContext.Instance
+
+		volName, volID := createAndValidateUniqueZonalDisk(client, p, z)
+
+		defer func() {
+			// Delete Disk
+			err := client.DeleteVolume(volID)
+			Expect(err).To(BeNil(), "DeleteVolume failed")
+
+			// Validate Disk Deleted
+			_, err = computeService.Disks.Get(p, z, volName).Do()
+			Expect(gce.IsGCEError(err, "notFound")).To(BeTrue(), "Expected disk to not be found")
+		}()
+
+		verifyVolumeStats := func(a verifyArgs) error {
+			available, capacity, used, inodesFree, inodes, inodesUsed, err := client.NodeGetVolumeStats(volID, a.publishDir)
+			if err != nil {
+				return fmt.Errorf("failed to get node volume stats: %v", err)
+			}
+			if !equalWithinEpsilon(available, common.GbToBytes(defaultSizeGb), defaultEpsiolon) || !equalWithinEpsilon(capacity, common.GbToBytes(defaultSizeGb), defaultEpsiolon) || !equalWithinEpsilon(used, 0, defaultEpsiolon) ||
+				inodesFree == 0 || inodes == 0 || inodesUsed == 0 {
```

Comment:
Should we write some files in the test so that inodes and used are non zero? (could be another test case)

Reply:
Also inodesFree is 0?

Reply:
dumb question - are there inodes in a raw block volume..?

Reply:
Inodes are a fs concept. This is the fs test case right?

Reply:
ah yes, this test is checking `if inodesFree==0 then Error`